index.rst

```diff
@@ -46,14 +46,314 @@
+Basically, the communication is broken down into a set of services, that are in charge of a particular aspect of the process.
+These services are highly configurable and it is even possible to run the the system with only a subset of them.
+
+These are the `OpenSplice DDS communication services`_:
+
+.. _OpenSplice DDS communication services: http://download.ist.adlinktech.com/docs/Vortex/html/ospl/DeploymentGuide/overview.html#vortex-opensplice-architecture
+
+  * `networking` - providing QoS-driven real-time networking based on multiple reliable multicast channels
+  * `durability` - providing fault-tolerant storage for both real-time state data as well as persistent settings
+  * `remote control and monitoring SOAP service` - providing remote web-based access using the SOAP protocol from various Vortex OpenSplice tools
+  * `dbms service` - providing a connection between the real-time and the enterprise domain by bridging data from DDS to DBMS and vice versa
+
+By default only the `networking` and `durability` services are configured.
+
+Furthermore, OpenSplice DDS can be configured to run in two modes of operation, :ref:`single process <fig-single-process>` or :ref:`shared memory <fig-shared-memory>`.
+When running in single process mode, the configured DDS services will run in the same process as the application.
+That means, alongside the resources needed by the application itself to run (computing, memory allocation, etc) there will also be additional resources needed by the DDS services themselves.
+In addition, there is also the question of how those additional resources are handled by the application and the programming language itself.
+For instance, Python does not easily support multiprocessing, so DDS services will be competing for the same computing resources as those needed by the application to run properly.
+
+.. figure:: /_static/SingleProcessFig.png
+   :name: fig-single-process
+   :target: ../_images/SingleProcessFig.png
+   :alt: DDS single process mode
+
+   DDS architecture when running in single process mode.
+   In the left hand panel, a diagram showing the internal architecture of a single application.
+   The "DDS Application Process" represents the software developed on top of SAL/SalObj to communicate over DDS and execute some operation.
+   In the case of Rubin Observatory system this would be a CSC, salkafka producer, user notebook server, etc.
+   In the right hand panel, a high level overview of a group of applications (CSCs, salkafka producers, etc).
+   Each application runs their own DDS services alongside their processes.
+
+As one can imagine, this architecture will not scale well for large scale systems.
+If the DDS communication grows large enough (in terms of number of topics and published frequency) the processing requirements for the DDS services alone may start to interfere with the application and vice-versa.
+To overcome this problem, OpenSplice DDS can be configured to run in shared memory mode.
```

Comment:
doesn't OpenSplice *recommened* to operate in shared memory mode?

Reply:
yes... for large scale system...